### DISCUSSION TITLE
Database.getConnection and jpaApi.withTransaction

### DISCUSSION THREAD

**Discussion by ppllaxxa** [playframework/playframework#11688] [Q&A] [answered]
Hello,

Could someone please help to clarify one thing?

Using jpaApi.withTransaction picking up one DB connection from the pull, it is clear.
The same for Database.getConnection and Database.withTransaction.

My question is - using below code - will it hold 2 different DB connections?
```
jpaApi.withTransaction(() -> {
  Connection conn = database.getConnection();
  doDBWork(conn);
})
```
or
```
jpaApi.withTransaction(() -> {
  database.withTransaction(conn -> {
    doDBWork(conn);
  });
})
```

Just found it in one of projects and per my understanding it will hold 1 additional DB connection - which means jast wasting time for 1 DB connection.

I could realize some rare cases where it could be usefule: some complex work in first transaction and in inner transaction we need to save some data to DB regardless of success for first transaction.

**Comment by mkurz** [comment, marked as answer]
Yes there will be two connections open this your examples.
